github.go

```diff
@@ -164,15 +164,37 @@ func getCommits(issueable Issueable, pullRequests PullRequests) ([]*github.Repos
 			message := fmt.Sprintf("Getting commits for PR %s failed", issue.FullName())
 			return nil, &ErrorResponse{err, http.StatusBadGateway, message}
 		}
+		isLastPage := resp.NextPage == 0
+		if isLastPage && !isExpectedHead(*lastCommit(pageCommits).SHA) {
+			message := fmt.Sprintf(
+				"Getting commits for PR %s failed. Received an unexpected head with SHA of %s.",
+				issue.FullName(),
+				*lastCommit(pageCommits).SHA,
+			)
+			if nrOfRetriesLeft <= 0 {
+				return nil, &ErrorResponse{nil, http.StatusBadGateway, message}
+			}
+			log.Printf("%s. Trying again.\n", message)
+			// Retry from page 1. This means clearing all the existing commits
+			// and restoring the page counter to 1.
+			commits = []*github.RepositoryCommit{}
+			pageNr = 1
+			nrOfRetriesLeft = nrOfRetriesLeft - 1
```

Comment:
I thought it wasn't supported, but turns out I was misremembering this thing: https://golang.org/doc/faq#inc_dec